identity/aziot-identityd/src/main.rs

```diff
@@ -60,26 +70,23 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
 			if prev_device_info_path.exists() {
 				let prev_hub_device_info = aziot_identityd::settings::HubDeviceInfo::new(&prev_device_info_path)?;
 
-				if let Some(prev_state) = prev_hub_device_info {
-					let mut prev_module_set: std::collections::BTreeSet<aziot_identity_common::ModuleId> = std::collections::BTreeSet::default();
-					if prev_state.eq(&curr_hub_device_info) && prev_settings_path.exists() {
-						let settings = aziot_identityd::settings::Settings::new(&prev_settings_path)?;
-						let (_, p) = convert_to_map(&settings.principal);
-						prev_module_set = p;
+				if let Some(prev_device) = prev_hub_device_info {
+					if prev_device == curr_hub_device_info {
+						let () = server_.init_hub_identities(prev_hub_mset, hub_mset).await?;
+						log::info!("Identity reconciliation with IoT Hub complete.");
 					}
-
-					let () = server_.init_identities(prev_module_set, module_set).await?;
 				}
 			}
 
-			log::info!("Identity reconciliation with IoT Hub complete.");
-
 			toml::to_string(&curr_hub_device_info)?
 		}
 		else {
 			aziot_identityd::settings::HubDeviceInfo::unprovisioned()
 		};
 
+		let () = server_.init_local_identities(prev_local_mset, local_mset).await?;
```

Comment:
I cannot review this logic until I see how you are handling reconciliation in init_local_identities. I think you need to be aware of your EST endpoint before reconciling your local identities on it. Otherwise, identity service will clobber all previous created identities after a server switch (which is likely not what we want).

Reply:
Pushed the commits with the reconciliation logic.